### PR TITLE
Updated IngestService to use new UniversalMetricFrame

### PR DESCRIPTION
**Pull request by rohitsngh27**
# Resolves
https://jira.rax.io/browse/SALUS-1004?

# What
Update IngestService to use UniversalMetricFrame schema

# How
Changes is IngestService class to use new UniversalMetricFrameSchema

# How to test
This can be tested via unit test cases

### CODE REVIEW

File: src/main/java/com/rackspace/salus/event/ingest/services/IngestService.java [hunk 1]
```diff
@@ -104,19 +105,21 @@ public void consumeMetric(ExternalMetric metric) {
...
-    if (StringUtils.hasText(metric.getDeviceLabel())) {
-      pointBuilder.tag(Tags.RESOURCE_LABEL, metric.getDeviceLabel());
-    }
     pointBuilder.tag(Tags.MONITORING_SYSTEM, metric.getMonitoringSystem().toString());
 
-    metric.getIvalues().forEach(pointBuilder::addField);
-    metric.getFvalues().forEach(pointBuilder::addField);
-    metric.getSvalues().forEach(pointBuilder::addField);
-
+    metric.getMetricsList().stream().forEach(val -> {
```
Comment (rohitsngh27) [inline src/main/java/com/rackspace/salus/event/ingest/services/IngestService.java:114]: Not sure about this logic here.

File: src/main/java/com/rackspace/salus/event/ingest/services/IngestService.java [hunk 2]
```diff
@@ -93,9 +94,9 @@ public void consumeMetric(ExternalMetric metric) {
     // Kapacitor provides a write endpoint that is compatible with InfluxDB, which is why
     // a native InfluxDB client is used here.
     final InfluxDB kapacitorWriter = kapacitorConnectionPool.getConnection(engineInstance);
-
-    final Instant timestamp = Instant.parse(metric.getTimestamp());
-    final Builder pointBuilder = Point.measurement(metric.getCollectionName())
+    Timestamp protobufTimestamp = metric.getMetrics(0).getTimestamp();
```
Comment (itzg) [inline src/main/java/com/rackspace/salus/event/ingest/services/IngestService.java:97]: A similar TODO comment would be good here

File: src/main/java/com/rackspace/salus/event/ingest/services/IngestService.java (hunk 1 above)
Comment (itzg) [reply src/main/java/com/rackspace/salus/event/ingest/services/IngestService.java:114]: Protobuf generates a `get{field}Case` method to determine which of the `oneof` fields is set:

https://developers.google.com/protocol-buffers/docs/reference/java-generated#oneof

For this one, the method is `getValueCase()`

**Review by itzg — CHANGES_REQUESTED** [review]

File: src/main/java/com/rackspace/salus/event/ingest/services/IngestService.java [hunk 3]
```diff
@@ -94,6 +96,7 @@ public void consumeMetric(UniversalMetricFrame metric) {
     // Kapacitor provides a write endpoint that is compatible with InfluxDB, which is why
     // a native InfluxDB client is used here.
     final InfluxDB kapacitorWriter = kapacitorConnectionPool.getConnection(engineInstance);
+    // TODO adjust recipient selection to handle non-Salus metrics
```
Comment (itzg) [inline src/main/java/com/rackspace/salus/event/ingest/services/IngestService.java:99]:
```suggestion
    // TODO adjust timestamp, measurement, and tags to handle non-Salus metrics
```

File: src/main/java/com/rackspace/salus/event/ingest/services/IngestService.java [hunk 4]
```diff
@@ -112,12 +115,16 @@ public void consumeMetric(UniversalMetricFrame metric) {
...
 
     metric.getMetricsList().stream().forEach(val -> {
-      if(!StringUtils.isEmpty(val.getString())) {
-        pointBuilder.addField(val.getName(), val.getString());
-      } else if(val.getFloat()!=0.0) {
-        pointBuilder.addField(val.getName(), val.getFloat());
-      } else if(val.getInt()!=0) {
-        pointBuilder.addField(val.getName(), val.getInt());
+      Metric.ValueCase valueCase = Metric.ValueCase.forNumber(val.getValueCase().getNumber());
+      switch (valueCase) {
```
Comment (itzg) [inline src/main/java/com/rackspace/salus/event/ingest/services/IngestService.java:119]: `val.getValueCase()` seems to return a `Metric.ValueCase` so just this should work:
```suggestion
      switch (val.getValueCase()) {
```

File: src/main/java/com/rackspace/salus/event/ingest/services/IngestService.java [hunk 5]
```diff
@@ -112,12 +115,16 @@ public void consumeMetric(UniversalMetricFrame metric) {
...
+        case INT:
+          pointBuilder.addField(val.getName(), val.getInt());
+          break;
+        case FLOAT:
+          pointBuilder.addField(val.getName(), val.getFloat());
+          break;
+        case STRING:
+          pointBuilder.addField(val.getName(), val.getString());
```
Comment (itzg) [inline src/main/java/com/rackspace/salus/event/ingest/services/IngestService.java:127]: To be safe, like if a new case is added later:
```suggestion
        case STRING:
          pointBuilder.addField(val.getName(), val.getString());
          break;
```

**Review by itzg — CHANGES_REQUESTED** [review]

**Review by itzg — APPROVED** [review]

**Pull request merged**